lib/iris/tests/unit/fileformats/netcdf/test__load_aux_factory.py

```diff
@@ -110,13 +110,38 @@ def test_formula_terms_p0_bounded(self):
             self.assertEqual(msg, str(warn[0].message))
 
     def test_formula_terms_ap_missing_coords(self):
-        coordinates = [(mock.sentinel.b, 'b'), (mock.sentinel.ps, 'ps')]
-        self.provides = dict(coordinates=coordinates)
         self.requires['formula_terms'] = dict(ap='ap', b='b', ps='ps')
         with mock.patch('warnings.warn') as warn:
             _load_aux_factory(self.engine, self.cube)
         warn.assert_called_once_with("Unable to find coordinate for variable "
                                      "'ap'")
+        # Check cube.add_aux_coord method.
+        self.assertEqual(self.cube.add_aux_coord.call_count, 0)
+        # Check cube.add_aux_factory method.
+        self.assertEqual(self.cube.add_aux_factory.call_count, 1)
+        args, _ = self.cube.add_aux_factory.call_args
+        self.assertEqual(len(args), 1)
+        factory = args[0]
+        # Check that the factory has no delta term
+        self.assertEqual(factory.delta, None)
+        self.assertEqual(factory.sigma, mock.sentinel.b)
+        self.assertEqual(factory.surface_air_pressure, mock.sentinel.ps)
+
+    def test_formula_terms_no_delta_terms(self):
+        self.requires['formula_terms'] = dict(b='b', ps='ps')
```

Comment:
Is it worth also adding a test for 
```self.requires['formula_terms'] = dict(a='a', b='b', ps='ps')```
```self.requires['formula_terms'] = dict(p0='p0', b='b', ps='ps')```

It then checks [this logic](https://github.com/SciTools/iris/pull/2485/files#diff-e055d3568b9b3f9cfe397cec7e016700R608)